# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: 09921685a632
# Full Hash: 09921685a632ade2b9da3e027f433c12902bccd3
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-08-04 21:52:46
# Description:
#   Bug 1656980 - CookieService::GetCookiesSince needs to check the CookieService initialization state, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D85824
# ==============================================================================

diff -r a6c4a0a21166 -r 09921685a632 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Tue Aug 04 10:29:38 2020 +0000
+++ b/netwerk/cookie/CookieService.cpp	Tue Aug 04 10:01:09 2020 +0000
@@ -2237,6 +2237,10 @@
 NS_IMETHODIMP
 CookieService::GetCookiesSince(int64_t aSinceWhen,
                                nsTArray<RefPtr<nsICookie>>& aResult) {
+  if (!IsInitialized()) {
+    return NS_OK;
+  }
+
   mPersistentStorage->EnsureReadComplete();
 
   // We expose only non-private cookies.
